controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -99,6 +104,37 @@ func (a *Adapter) EnsureIntegrationTestPipelineForScenarioExists() (reconciler.O
 	return reconciler.ContinueProcessing()
 }
 
+// EnsureEphemeralEnvironmentsCleanedUp will ensure that ephemeral environment(s) associated with the
+// SnapshotEnvironmentBinding are cleaned up.
+func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (reconciler.OperationResult, error) {
+
+	if !gitops.HaveBindingsFailed(a.snapshotEnvironmentBinding) {
+		return reconciler.ContinueProcessing()
+	}
+
+	// mark snapshot as failed
+	_, err := gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot, "Encountered issue deploying snapshot on ephemeral environments")
```

Comment:
Instead of hardcoding the failure reason of the Snapshot, what do we think about copying over the content of the `message` field from the `BindingErrorOccurredStatusConditionType` of SEB?